### PR TITLE
Expand header_widget to include doc path, citation and help text

### PR DESCRIPTION
**Pull request by K-Meech**
## Description

**What is this PR**

- [ ] Bug fix
- [X] Addition of a new feature
- [ ] Other

**Why is this PR needed?**
The `header_widget` currently doesn't support some of the required features for existing brainglobe headers like [cellfinder](https://github.com/brainglobe/cellfinder/blob/main/cellfinder/napari/utils.py) and [brainrender-napari](https://github.com/brainglobe/brainrender-napari/blob/main/brainrender_napari/utils/brainglobe_logo.py)

**What does this PR do?**
This PR adds new features to the `header_widget`:
- Makes tutorial links optional
- Adds option for documentation links
- Adds option for citation doi
- Adds option for help text
- Adds option for github repo name (if not the same as the package name)

## References

For https://github.com/brainglobe/cellfinder/issues/327

## How has this PR been tested?

All tests pass locally.

## Is this a breaking change?

No

## Does this PR require an update to the documentation?

Relevant docstrings have been updated.

## Checklist:

- [X] The code has been tested locally
- [X] Tests have been added to cover all new functionality (unit & integration)
- [X] The documentation has been updated to reflect any changes
- [X] The code has been formatted with [pre-commit](https://pre-commit.com/)

### CODE REVIEW

File: brainglobe_utils/qtpy/logo.py [hunk 1]
```diff
@@ -6,18 +6,51 @@
 def _docs_links_widget(
     package_name: str,
     package_tagline: str,
-    tutorial_file_name: str,
+    tutorial_file_name: str = None,
+    documentation_path: str = None,
+    citation_doi: str = None,
+    github_repo_name: str = None,
+    help_text: str = None,
```
Comment (willGraham01) [inline brainglobe_utils/qtpy/logo.py:13]:
```suggestion
    tutorial_file_name: Optional[str] = None,
    documentation_path: Optional[str] = None,
    citation_doi: Optional[str] = None,
    github_repo_name: Optional[str] = None,
    help_text: Optional[str] = None,
```

Will also need to `from typing import Optional` at the top of the file.

File: brainglobe_utils/qtpy/logo.py [hunk 2]
```diff
@@ -38,9 +71,13 @@ def _logo_widget(package_name: str, parent: QWidget = None):
 def header_widget(
     package_name: str,
     package_tagline: str,
-    tutorial_file_name: str,
+    tutorial_file_name: str = None,
+    documentation_path: str = None,
+    citation_doi: str = None,
+    github_repo_name: str = None,
+    help_text: str = None,
```
Comment (willGraham01) [inline brainglobe_utils/qtpy/logo.py:78]:
```suggestion
    tutorial_file_name: Optional[str] = None,
    documentation_path: Optional[str] = None,
    citation_doi: Optional[str] = None,
    github_repo_name: Optional[str] = None,
    help_text: Optional[str] = None,
```

**Comment by K-Meech** [comment]
Example usage for `cellfinder` [in this draft PR](https://github.com/brainglobe/cellfinder/pull/399)

File: brainglobe_utils/qtpy/logo.py [hunk 3]
```diff
@@ -51,9 +88,21 @@ def header_widget(
         The name of the package, e.g. "brainrender-napari"
     package_tagline : str
         The tagline for the package
-    tutorial_file_name : str
+    tutorial_file_name : str, optional
         The name of the tutorial file (must include .html),
-         e.g. "brainrender-qtpy.html": str
+         e.g. "brainrender-qtpy.html"
+    documentation_path : str, optional
+        The relative path of a documentation file (must include .html),
```
Comment (willGraham01) [inline brainglobe_utils/qtpy/logo.py:95]: > The relative path of a documentation file (must include .html),

I'd include what this is relative to (it looks like it's relative to `https://brainglobe.info/documentation/` from inspection of the `_docs_links_widget`) in the docstring itself.

**Review by willGraham01 — APPROVED** [review]
Just a few minor docstring things, and technically `PEP` wants us to include the `Optional` keyword in the typehints (personally I still believe it's overkill but the `PEP` overlords have spoken).

**Comment by K-Meech** [comment]
All fixed now @willGraham01 ! Hopefully the `PEP` overlords are satisfied :laughing:. Feel free to merge if you're happy with it

**Pull request merged**